core/vision/src/main/java/com/arcrobotics/ftclib/vision/UGAdvancedHighGoalPipeline.java

```diff
@@ -65,30 +74,21 @@ public double getPowershotAngle(Target color, Powershot shot) {
         double distanceFromWallToGoal = Math.tan(Math.toRadians(angle)) * getDistanceToGoalWall(color);
 
         // If robot is inbetween Powershot and Highgoal
-        if (Math.abs(offset) > Math.abs(distanceFromWallToGoal)) {
-            return Math.toRadians(Math.atan((offset - distanceFromWallToGoal) / getDistanceToGoalWall(color)));
+        if (Math.abs(offset) > Math.abs(distanceFromWallToGoal) && (offset * distanceFromWallToGoal < 0)) {
+            return Math.toDegrees(Math.atan((offset + distanceFromWallToGoal) / getDistanceToGoalWall(color)));
         }
 
-        // if the angle to the powershot is embedded in the angle to the color goal
-        boolean inclusiveAngle = angle * offset < 0;
-        // triangle made from distangeToGoalWall and angle
-        double hypotenuse = getDistanceToGoalWall(color) / Math.cos(Math.toRadians(angle));
-        if (inclusiveAngle) {
-            return angle - Math.toDegrees(Math.asin(offset / hypotenuse));
+        if (angle > 0) {
+            return Math.toDegrees(Math.atan((offset + distanceFromWallToGoal) / getDistanceToGoalWall(color)));
         } else {
-            double alpha = Math.toDegrees(Math.atan((Math.abs(offset) + Math.abs(distanceFromWallToGoal)) / getDistanceToGoalWall(color))) - angle;
-            return (Math.signum(angle) * alpha) + angle;
+            return Math.toDegrees(Math.atan((offset - distanceFromWallToGoal) / getDistanceToGoalWall(color)));
         }
 
 
```

Comment:
Remove the white space here. The closing brace for the method should be flush with the closing brace of the if-else statement.